src/BoundingBoxCommand.cc

```diff
@@ -0,0 +1,370 @@
+/**
+ * @file   BoundingBoxCommand.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+#include <iostream>
+
+#include "BoundingBoxCommand.h"
+#include "VDMSConfig.h"
+#include "defines.h"
+
+using namespace VDMS;
+
+//========= RegionCommand definitions =========
+
+BoundingBoxCommand::BoundingBoxCommand(const std::string &cmd_name):
+    RSCommand(cmd_name)
+{
+}
+
+//========= AddBoundingBox definitions =========
+
+AddBoundingBox::AddBoundingBox() : BoundingBoxCommand("AddBoundingBox")
+{
+}
+
+int AddBoundingBox::construct_protobuf(PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    int node_ref = get_value<int>(cmd, "_ref",
+                                  query.get_available_reference());
+
+    Json::Value rect = get_value<Json::Value>(cmd, "rectangle", Json::Value::null);
+    Json::Value props = get_value<Json::Value>(cmd, "properties");
+
+    if (rect != Json::Value::null) {
```

Comment:
That's left from when polygons were also an option, since it was either rectangle or polygon. I can take it out, but I don't think it's going to hurt to leave it in either?

Reply:
It is harmless so its ok but typically we have been trying to remove code that is not relevant to the current commit.

Reply:
I can take it out, it's not a big deal